stl.go

```diff
@@ -628,15 +698,20 @@ func (t *ttiBlock) bytes(g *gsiBlock) (o []byte) {
 	o = append(o, byte(uint8(t.subtitleGroupNumber))) // Subtitle group number
 	var b = make([]byte, 2)
 	binary.LittleEndian.PutUint16(b, uint16(t.subtitleNumber))
-	o = append(o, b...)                                                                                              // Subtitle number
-	o = append(o, byte(uint8(t.extensionBlockNumber)))                                                               // Extension block number
-	o = append(o, t.cumulativeStatus)                                                                                // Cumulative status
-	o = append(o, formatDurationSTLBytes(t.timecodeIn, g.framerate)...)                                              // Timecode in
-	o = append(o, formatDurationSTLBytes(t.timecodeOut, g.framerate)...)                                             // Timecode out
-	o = append(o, byte(uint8(t.verticalPosition)))                                                                   // Vertical position
-	o = append(o, t.justificationCode)                                                                               // Justification code
-	o = append(o, t.commentFlag)                                                                                     // Comment flag
-	o = append(o, astikit.BytesPad(encodeTextSTL(string(t.text)), '\x8f', 112, astikit.PadRight, astikit.PadCut)...) // Text field
+	o = append(o, b...)                                                  // Subtitle number
+	o = append(o, byte(uint8(t.extensionBlockNumber)))                   // Extension block number
+	o = append(o, t.cumulativeStatus)                                    // Cumulative status
+	o = append(o, formatDurationSTLBytes(t.timecodeIn, g.framerate)...)  // Timecode in
+	o = append(o, formatDurationSTLBytes(t.timecodeOut, g.framerate)...) // Timecode out
+	o = append(o, byte(uint8(t.verticalPosition)))                       // Vertical position
+	o = append(o, t.justificationCode)                                   // Justification code
+	o = append(o, t.commentFlag)                                         // Comment flag
+	//if text has 112 characters, copy directly
+	if len(t.text) == 112 {
```

Comment:
Remove this micro-optimization as you need to compare `len(encodeTextSTL(string(t.text))` anyway

Reply:
If the text is from STL to STL, we dont need to encode. We need to mantain spaces, colour, height, font style that comes in text field

Reply:
What you're saying makes sense, however I don't like basing the "not encoding" part on the length of `t.text`.

Out of simplicity, I'd rather:

- use `encodeTextSTL` [here](https://github.com/asticode/go-astisub/pull/25/files#diff-50f54403757419c518c555eeadf53396R675) (that way, if `t.text` comes from `i.Metadata.STLText` it's never encoded and your spaces, colour, height, etc. are maintained)
- remove the `if len(t.text) == 112` and partially revert to 

```go
o = append(o, astikit.BytesPad(t.text), '\x8f', 112, astikit.PadRight, astikit.PadCut)...)
```

Reply:
I'm testing this and the codification changes if we use encodeTextSTL.
Here the example:
![ErrorCodificationSTL](https://user-images.githubusercontent.com/8900061/91836021-82896980-ec4a-11ea-8026-b0f1f90fd020.png)

Reply:
This doesn't surprise me since there's [this TODO](https://github.com/asticode/go-astisub/blob/master/stl.go#L687).

However that doesn't change anything for you, if you move `encodeTextSTL` [here](https://github.com/asticode/go-astisub/pull/25/files#diff-50f54403757419c518c555eeadf53396R675) in your case `encodeTextSTL` will never get called since you're using `i.Metadata.STLText` and therefore codification won't change.

I think this is best for this PR to make this change, and removing `encodeTextSTL` in another PR.

Reply:
I don't understand where I should move encodeTextSTL